source/server/overload_manager_impl.h

```diff
@@ -52,6 +52,49 @@ class OverloadAction {
   Stats::Gauge& scale_percent_gauge_;
 };
 
+// Simple table that converts strings into Symbol instances. Symbols are guaranteed to start at 0
+// and be indexed sequentially.
+class NamedOverloadActionSymbolTable {
+public:
+  class Symbol {
+  public:
+    // Allow copy construction everywhere.
+    Symbol(const Symbol&) = default;
+
+    // Returns the index of the symbol in the table.
+    size_t index() const { return index_; }
+
+    // Support use as a map key.
+    bool operator==(const Symbol other) { return other.index_ == index_; }
+    template <typename H> friend H AbslHashValue(H h, const Symbol& s) {
+      return H::combine(std::move(h), s.index_);
+    }
+
+  private:
+    friend class NamedOverloadActionSymbolTable;
+    // Only the symbol table class can create Symbol instances from indicies.
+    explicit Symbol(size_t index) : index_(index) {}
+
+    size_t index_;
+  };
+
+  // Finds an existing or adds a new entry for the given name.
+  Symbol get(absl::string_view name);
```

Comment:
Ah, now I understand the naming convention. Maybe this should be `SymbolTable::LookupOrInsert()` and `Find()` should just be `Lookup()`? I don't feel very strongly about this, though.